lib/yggdrasil/plug/authenticate.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Yggdrasil.Plug.Authenticate do
+  import Plug.Conn
+  import Yggdrasil.Router.Helpers
+  import Phoenix.Controller
+
+  require Logger
+
+  def init(default), do: default
+
+  def call(conn, default) do
```

Comment:
This can absolutely be tested and I was reading on it, still not exactly sure what to include but I know there helpers for a connection. So we should be able to test that it redirected and where of if succeded that it assigned the user and the token.